### PR TITLE
Filters, location filter and garden filter

### LINKED ISSUE #61: Add 'search' tool to type location
Suggestion of user:

Add search tool so you can also search on names of places, next to the map-zoom-search.

### PR DESCRIPTION
**Pull request by wardbeyens**
closes #61

video: https://cloud.wabyte.com/index.php/s/JsmdbBdb5xxEJtq

The user types in the input field, if the user stops typing for 250 milliseconds then there are (normally) 5 results, i.e. 5 place names displayed below the input field.
These results appear and disappear with a fade-in (50ms) and fade-out (200ms).

If the user clicks on 1 of the results (i.e. 1 location) the map will zoom in to this location. The input field will be cleared and the results will disappear.

Currently the search is limited to cities and villages and the results are displayed in the language of the user.

If the user searches for a location and none is found a message "no places found" will be displayed for 3 seconds.

If the results are still on the screen, but the user clicks not on one of the locations, but on the map. Then the input field will be emptied as the results. (When the user drags the map, the locations WILL be displayed)

Future possibilities:
- accessibility, 'tab' 'shift tab' 'arrow up' 'arrow down' 'enter' => go over the places list, enter for selecting the place.
- dynamic zoom, search for Bruxelles zoom=10, search for small village zoom= 13 https://wiki.openstreetmap.org/wiki/Zoom_levels

known issue: 
- phone, while typing the search results are displayed over the trails box and not all the results are visible. (don't know how to fix)
link image not typing: https://cloud.wabyte.com/index.php/s/c3PKi7bPSPfwTsR
link image typing: https://cloud.wabyte.com/index.php/s/mK5K78L5MYyH84d

### CODE REVIEW

**Comment by MichielLeyman** [comment]
@wardbeyens we can't merge this without the filter button being finished right? Is this a draft?

**Comment by wardbeyens** [comment]
Yes this is supposed to be a draft

**Comment by wardbeyens** [comment]
closes #51

**Comment by wardbeyens** [comment]
@MichielLeyman 
The filter feature is ready.
Both the location filter and the garden filter have been tested on different platforms and seem to work well.

**Comment by wardbeyens** [comment]
> @wardbeyens we can't merge this without the filter button being finished right? Is this a draft?

Finished :smile:

**Comment by wardbeyens** [comment]
The last 2 commits ensure that the map is centred on a user's garden, IP address, user's country with Belgium as fallback.

**Comment by MichielLeyman** [comment]
Closes #55

**Comment by MichielLeyman** [comment]
Closes #149

File: src/components/Garden/Filter.svelte [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+          <p id="capacity">{filter.capacity.min}</p>
+          <button on:click={capacityMinIncrease}>+</button>
+        </div>
+      </div>
+    </div>
+  </div>
+  <span slot="controls" class="applyGardenFilter">
+    <p class="light">{Object.values(filteredGardens).length} gardens</p>
+    <Button on:click={() => filterGardens()} medium uppercase>Apply filter</Button>
+  </span>
```
Comment (MichielLeyman) [inline src/components/Garden/Filter.svelte]: These still need translation

File: src/components/Garden/FilterLocation.svelte [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        {displayPlaceName(place.place_name)}
+      </button>
+      {#if i != places.length - 1}
+        <hr />
+      {/if}
+    {/each}
+  </div>
+{:else if showNoPlacesBool}
+  <div class="location-filter-output-error">
+    <p>No places found</p>
```
Comment (MichielLeyman) [inline src/components/Garden/FilterLocation.svelte:133]: This still needs translation

File: src/components/UI/Modal.svelte [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            <Icon icon={crossIcon} />
+          </button>
+        {/if}
+      </div>
+      <div class="modal-body">
+        <slot name="body" {ariaLabelledBy} {ariaDescribedBy} />
+      </div>
+      <div class="controls">
+        {#if cancelButton}
+          <Button type="button" uppercase inverse on:click={close}>Close</Button>
```
Comment (MichielLeyman) [inline src/components/UI/Modal.svelte:78]: This still needs translation

**Review by MichielLeyman — CHANGES_REQUESTED** [review]
@wardbeyens I'm working on getting this PR out the door finally, sorry for the delay on my end. When you have a spare moment, could you get the missing translations in (marked below) while I work on the location filtering and centering?

File: src/components/Garden/Filter.svelte (hunk 1 above)
Comment (wardbeyens) [reply src/components/Garden/Filter.svelte]: damn i missed this one, will do!

**Pull request closed without merging**